### ISSUE TITLE
Can't built main

### ISSUE THREAD

**Issue by buffalu** [blockworks-foundation/mango-v3#84]
Sup mango team.

I can't build on main right now, seems like something wrong with switchboard dependency.

```
   Compiling switchboard-utils v0.1.29
error[E0080]: evaluation of constant value failed
  --> /Users/.../.cargo/registry/src/github.com-1ecc6299db9ec823/switchboard-utils-0.1.29/src/lib.rs:19:18
   |
19 | #[derive(Default,AsBytes,FromBytes,Clone,Debug)]
   |                  ^^^^^^^ attempt to divide `1_usize` by zero
   |
   = note: this error originates in the derive macro `AsBytes` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0080`.
error: could not compile `switchboard-utils` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed

mango-v3 % rustup --version
rustup 1.24.3 (ce5817a94 2021-05-31)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.59.0-nightly (207c80f10 2021-11-30)`

% rustc --version
rustc 1.59.0-nightly (207c80f10 2021-11-30)
```

**Comment by dafyddd** [comment]
I think the solana compiler comes with a rustc. And I use solana-cli 1.8 to compile and don't have any issues.

**Comment by thanhngvpt** [comment]
same error, I also tried with rustc 1.57.0-nightly (2021-09-07) same error. What wrong with me?

**Comment by microwavedcola1** [comment]
@thanhngvpt @buffalu are you guys using rosetta compatibility mode? m1 is my main driver, and havent had any issues so far

**Comment by RohanKapurDEV** [comment]
havent had any issues building on m1 either

**Comment by sthornington** [comment]
The root cause of this is that mango (and switchboard) have `#[repr(C)]` for some account data structs (SwitchboardDecimal, *Node in matching.rs and *Event in queue.rs), but on ARM, the alignment of `i128` is 16 bytes, vs 8 bytes for x86_64.  These structs should all be `#[repr(packed)]` in my opinion, as they are in the serum dex

**Comment by sthornington** [comment]
In fact, it looks like SwitchboardDecimal is already packed in the latest...

**Comment by sthornington** [comment]
@ckamm did you have any luck upgrading switchboard?  I see you pinned the versions to older ones fairly recently...

**Comment by ckamm** [comment]
Upgrading switchboard wasn't a priority so I haven't looked at it further. This PR has an upgrade though: https://github.com/blockworks-foundation/mango-v3/pull/150

**Comment by sthornington** [comment]
Gotcha. Yeah I haven’t gotten further enough to see whether switching all the account state structs to repr(packed) fixes everything else. I suspect one day everyone will be forced to switch to proper serdes using borsh rather than bytemuck casting.

**Comment by smu160** [comment]
I ran into the same issue, and I was able to rectify the issue with `rustup` by switching the active tool-chain.

You can confirm by running: `rustup show`. If you see `stable-x86_64-apple-darwin (default)` under `active toolchain`, you're good to go.

Otherwise, do as follows:

```
rustup default stable-x86_64-apple-darwin
```

Try it out:
```
cargo build
```

You should be able to build without any platform specific issues now.

**Comment by sthornington** [comment]
Yes of course you can fix an issue with bad alignment on ARM by not compiling on ARM. 😀

**Comment by namnh06** [comment]
I've got the same issue with Ubuntu 20.04
![image](https://user-images.githubusercontent.com/28762252/163832661-4db9d3cb-dc2c-41f4-a322-e2415bf2271e.png)
![image](https://user-images.githubusercontent.com/28762252/163832709-bae77b20-2840-4374-b1df-0c85ee54b325.png)
